lumicks/pylake/tests/test_kymo.py

```diff
@@ -266,7 +266,82 @@ def test_downsampled_kymo():
         kymo_ds.timestamps
 
     # Verify that we can pass a different reduce function
-    assert np.allclose(kymo.downsampled_by(2, reduce=np.mean).red_image, ds / 2)
+    assert np.allclose(kymo.downsampled_by(time_factor=2, reduce=np.mean).red_image, ds / 2)
+
+
+def test_downsampled_kymo_position():
+    """Test downsampling over the spatial axis"""
+    image = np.array(
+        [
+            [0, 12, 0, 12, 0, 6, 0],
+            [0, 0, 0, 0, 0, 6, 0],
+            [12, 0, 0, 0, 12, 6, 0],
+            [0, 12, 12, 12, 0, 6, 0],
+            [0, 12, 12, 12, 0, 6, 0],
+        ],
+        dtype=np.uint8
+    )
+
+    kymo = generate_kymo(
+        "Mock", image, pixel_size_nm=1, start=100, dt=5, samples_per_pixel=5, line_padding=2
+    )
+
+    kymo_ds = kymo.downsampled_by(position_factor=2)
+    ds = np.array([[0, 12, 0, 12, 0, 12, 0], [12, 12, 12, 12, 12, 12, 0]], dtype=np.uint8)
+    ds_ts = np.array([[132.5,  277.5,  422.5,  567.5,  712.5,  857.5, 1002.5],
+                      [182.5,  327.5,  472.5,  617.5,  762.5,  907.5, 1052.5]])
+
+    assert kymo_ds.name == "Mock"
+    assert np.allclose(kymo_ds.red_image, ds)
+    assert np.allclose(kymo_ds.timestamps, ds_ts)
+    assert np.allclose(kymo_ds.start, 100)
+    assert np.allclose(kymo_ds.pixelsize_um, 2 / 1000)
+    assert np.allclose(kymo_ds.line_time_seconds, kymo.line_time_seconds)
+
+    # We lost one line while downsampling
+    assert np.allclose(kymo_ds.size_um[0], kymo.size_um[0] - kymo.pixelsize_um[0])
+
+    # Verify that we can pass a different reduce function
+    alt_ds = kymo.downsampled_by(position_factor=2, reduce=np.mean, reduce_timestamps=np.sum)
+    assert np.allclose(alt_ds.red_image, ds / 2)
+    assert np.allclose(alt_ds.timestamps, ds_ts * 2)
+
+
+def test_downsampled_kymo_both_axes():
+    image = np.array(
+        [
+            [0, 12, 0, 12, 0, 6, 0],
+            [0, 0, 0, 0, 0, 6, 0],
+            [12, 0, 0, 0, 12, 6, 0],
+            [0, 12, 12, 12, 0, 6, 0],
+            [0, 12, 12, 12, 0, 6, 0],
+        ],
+        dtype=np.uint8
+    )
+
+    kymo = generate_kymo(
+        "Mock", image, pixel_size_nm=1, start=100, dt=5, samples_per_pixel=5, line_padding=2
+    )
+    ds = np.array([[12, 12, 12], [24, 24, 24]], dtype=np.uint8)
+
+    downsampled_kymos = [
+        kymo.downsampled_by(time_factor=2, position_factor=2),
+        # Test whether sequential downsampling works out correctly as well
+        kymo.downsampled_by(position_factor=2).downsampled_by(time_factor=2),
+        kymo.downsampled_by(time_factor=2).downsampled_by(position_factor=2)
```

Comment:
I wouldn't have thought to test for function chaining. Good stuff! Perhaps could test that the original `kymo` instance is still working after downsampling and that `downsampled_by` has no side effects on the instance?

Reply:
Not a bad idea. Added as a new test!